stdlib/Unicode/src/Unicode.jl

```diff
@@ -89,4 +89,86 @@ letter combined with an accent mark is a single grapheme.)
 """
 graphemes(s::AbstractString) = Base.Unicode.GraphemeIterator{typeof(s)}(s)
 
+const emoji_data = download("https://www.unicode.org/Public/13.0.0/ucd/emoji/emoji-data.txt")
```

Comment:
I don’t think we want to store this entire string in the compiled library. You should just download it when you parse the data, maybe?

Reply:
Yeah, that makes sense.

Reply:
Your old code that just looked at certain code point blocks was a lot more compact and independent of the Unicode version. I’m just not sure if it is standard conforming?  

Could we just use the old code combined with checking the category code to see if the code point is assigned?

Reply:
(I still think we should use the data file, but only for tests.)

Reply:
As i looked more into the emoji codepoints, i found some corner cases which weren't well-covered by the old version. It would be possible to patch those, but the secondary issue was that there's a bunch of unassigned emoji in the blocks in the old version. If we're ok with saying "yes this is an emoji" to characters in those larger blocks which are not (yet) emoji, then we can go with the old system. If not, we would either need to restrict those ranges to those codepoints which are currently assigned manually or via parsing the emoji_data file like i do here.

i do agree the old version was a lot simpler, so i'm not sure what the best solution is.

Reply:
I agree that we should return `false` for currently unassigned codepoints, but we can check for that simply by returning `false` if `Unicode.category_code(char) == Unicode.UTF8PROC_CATEGORY_CN`

Reply:
Oh neat, didn't know about that. I'll get back with a new version later today.

Reply:
So looking at this, there are things in the emoji data file which fall under unicode category Sm and Po (◾ and ‼, respectively), and things in So which do not qualify as emoji (Ⓕ). To be honest, I'm not sure why some of these symbols get called emoji and some don't, but I think the only way to be complete here is to use the full list of ranges. We don't have to download it necessarily (could take the output from the file and manually include that in the unicode.jl file, but that's a 702-element array), but I'm not sure of another way to make sure we catch all emoji without false positives